doc-surrealdb_versioned_docs/version-1.x/surrealql/statements/define/field.mdx

```diff
@@ -276,4 +282,14 @@ DEFINE FIELD countrycode ON user TYPE string
 	-- Set a default value if empty
 	VALUE $value OR $before OR 'GBR'
 ;
+```
+
+## Interacting with the parent table
+
+The $this parameter gives access inside a DEFINE FIELD statement to the parent table.
```

Comment:
Maybe "The $this parameter inside a DEFINE FIELD statement refers to the table on which the field is defined"?

Or now that there is a separate Notable parameters section that already introduces it, maybe something more use case-focused like "The $this parameter is useful for accessing or making assertions on the fields of the table on which the field is defined." (And then into the example, maybe add another one showing ASSERT)

Reply:
Is there a reason to avoid the word `record` and use `table` in this situation? The moment a `table` is mentioned it instantly brings 2 things to my mind:
- The table schema definition
- All the records in the table
As far as I understand, the $this doesn't point to the whole table, but simply to a single record the operation is being performed on.

---

Having all parameter definitions solely in "Parameters" page sounds like a good idea. The things that the page needs are:
1. List all available contexts where each parameter can be used.
2. Point out how the parameter value differs in some contexts (in `ASSERT` the `$after` is the same as `$value` for example)
3. Some parameters could use an explicit statement that they are available in all queries, like `$session`.

It would be a lot to keep track though.

Reply:
That reminds me a bit of closures in Rust which technically are only closures if it captures (en"closes") an outside variable and otherwise should be called anonymous functions...but in practice everything with a || just gets called a closure for simplicity and to avoid shocking new users.

Similarly here I think table would make more sense because it's being used in the context of DEFINE FIELD / DEFINE TABLE and using the word record could confuse a new user to databases into thinking that there is a DEFINE RECORD or somesuch. Unless there is some clever wording that could refer to both at the same time.

There is a page devoted to parameters with examples for each, do you think it does the job? https://surrealdb.com/docs/surrealdb/surrealql/parameters Can always be expanded if necessary.

Reply:
I understand what you're saying. From my perspective the word `table` adds confusion as being within the `ASSERT` statement, the context is a concrete `field with some data` that belongs to a `concrete record`. The `table` is a parent of a `field definition` where there's no concrete data. It's an interesting problem.

I am aware of the `Parameters` page. My problem with the page is that currently it presents the parameters only in a very specific situation. For example the `$this` parameter is presented only within `CREATE` and `SELECT` statements. There's no mention of it being available (and what it represents) in `ASSERT` or any other statements.

It sounds like a lot but what if the `Parameters` page would be expanded, where each parameter gets a list of ALL possible places where it can be used. Each possible usage would mention:
- what the parameter stores in that specific situation
- clear example of usage

This would make the page LONG, but at the same time it's documentation and it could be split into multiple smaller pages. I'd be happy to contribute once there's a template established.

Reply:
I think I have an idea that may work, will split into two PRs and close this one (plus the docs structure was just redone yesterday too, much easier structure but the merge conflicts are out of this world). I'll tag you in a sec

Reply:
@Xkonti Here's the new PR for the DEFINE FIELD and DEFINE TABLE pages only: https://github.com/surrealdb/docs.surrealdb.com/pull/710